File: frontend/src/types/baccarat.ts
import { BaseGameState, BasePlayer, Card } from './game'

export interface BaccaratHand {
  cards: Card[]
  total: number
}

export interface BaccaratBets {
  player: number
  banker: number
  tie: number
}

export interface BaccaratPlayer extends BasePlayer {
  bets: BaccaratBets
}

export interface BaccaratGameState extends BaseGameState {
  gameType: 'baccarat'
  playerHand: BaccaratHand
  bankerHand: BaccaratHand
  players: BaccaratPlayer[]
  result?: 'player' | 'banker' | 'tie'
  chipChange?: number
}

File: backend/internal/models/baccarat.go
package models

type BaccaratHand struct {
	Cards []Card `json:"cards"`
	Total int    `json:"total"`
}

type BaccaratBets struct {
	Player int64 `json:"player"`
	Banker int64 `json:"banker"`
	Tie    int64 `json:"tie"`
}

type BaccaratPlayer struct {
	Player
	Bets BaccaratBets `json:"bets"`
}

type BaccaratGameState struct {
	BaseGameState
	PlayerHand BaccaratHand   `json:"playerHand"`
	BankerHand BaccaratHand   `json:"bankerHand"`
	Players    []BaccaratPlayer `json:"players"`
	Result     string         `json:"result,omitempty"` // player, banker, tie
	ChipChange int64          `json:"chipChange,omitempty"`
}
